lib/src/parsers/schemaprop-parser.ts

```diff
@@ -0,0 +1,264 @@
+import { JSDoc } from "ts-morph";
+import { SchemaProp, Type, TypeKind } from "../types";
+import { err, ok, Result } from "../util";
+import { ParserError } from "../errors";
+
+export function extractJSDocSchemaProps(
+  jsDoc: JSDoc | undefined,
+  type: Type
+): Result<SchemaProp[], ParserError> | undefined {
+  // return early if there is no jsDoc available
+  if (!jsDoc) {
+    return;
+  }
+
+  const rawDefaults = jsDoc
+    .getTags()
+    .filter(tag => tag.getTagName() === "default")
+    .map(schemaProp => schemaProp.getComment());
+  const parentJsDocNode = jsDoc.getParent();
+  if (rawDefaults && rawDefaults.indexOf(undefined) !== -1) {
+    return err(
+      new ParserError("Default must not be empty", {
+        file: parentJsDocNode.getSourceFile().getFilePath(),
+        position: parentJsDocNode.getPos()
+      })
+    );
+  }
+
+  const rawSchemaProps = jsDoc
+    .getTags()
+    .filter(tag => tag.getTagName() === "oaSchemaProp")
+    .map(schemaProp => schemaProp.getComment());
+  if (rawSchemaProps && rawSchemaProps.indexOf(undefined) !== -1) {
+    return err(
+      new ParserError("schemaProp must not be empty", {
+        file: parentJsDocNode.getSourceFile().getFilePath(),
+        position: parentJsDocNode.getPos()
+      })
+    );
+  }
+
+  if (rawDefaults && rawDefaults.length > 0) {
+    rawDefaults.every(defaultValue => {
+      rawSchemaProps.push("default\n" + defaultValue);
+    });
+  }
+
+  if (rawSchemaProps && rawSchemaProps.length > 0) {
+    const schemaProps: SchemaProp[] = [];
+    let schemaPropError;
+    const propTypeMap = new Map<
```

Comment:
We can move this `Map` to a constant outside the `extractJSDocSchemaProps` function

Reply:
ok, done